ADAL/src/cache/ios/ADKeychainTokenCache.m

```diff
@@ -395,28 +395,19 @@ - (BOOL)removeAllForClientId:(NSString * __nonnull)clientId
     AD_LOG_WARN_DICT(([NSString stringWithFormat:@"Removing all items for client <%@>", clientId]), nil,
                      (@{ @"operation" : @"removeAllForClientId:", @"clientId" : clientId }), nil);
     
-    BOOL deleteSuccessful = YES;
     NSArray* items = [self allItems:nil];
     
-    ADAuthenticationError* adError = nil;
-    
     for (ADTokenCacheItem * item in items)
     {
         if ([clientId isEqualToString:[item clientId] ])
         {
-            [self removeItem:item error:&adError];
-            if (adError)
+            if (![self removeItem:item error:error])
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1066)